Shopify API Client: update to latest version

danielbeardsley opened the pull request:
The old pagination method is being deprecated by shopify so we have to
keep up with the times.
                                                    
New client doesn't appear to need the API key anymore, just the password.

Note: also includes refreshing the Dockerfile and dealing with a security vuln
on a dependency.
                          
Closes iFixit/ifixit#34863

danielbeardsley committed:
Shopify API Client: update to latest version

The old pagination method is being deprecated by shopify so we have to
keep up with the times.

New client doesn't appear to need the API key anymore, just the password.

danielbeardsley committed:
Dockerfile: use official python image

danielbeardsley committed:
PyYML: address security issue

Dependabot opened this a year ago, let's bring this update
into this branch since we're testing it now.

danielbeardsley commented:
QA :+1: 

We've (@sctice and I ) have tested this inside and outside of docker and it works!

sterlinghirsh commented on get_orders.py:
```diff
@@ -46,22 +46,18 @@ def get_all_orders(orders_getter, limit=50):
-                                                         os.environ['SHOPIFY_SHOP_NAME'])
-shopify.ShopifyResource.set_site(shop_url)
+shop_url = "https://{}.myshopify.com/admin".format(os.environ['SHOPIFY_SHOP_NAME'])
+session = shopify.Session(shop_url, "2020-10", os.environ['SHOPIFY_PASSWORD'])
```
Why use 2020-10? Looks like the current version is 2021-01. Maybe this should be an environment var?

danielbeardsley replied:
Good point. Fixed.

danielbeardsley committed:
Dockerfile: utilize WORKDIR

It makes the directory *and* changes the current dir foa ll successive
commands.

danielbeardsley committed:
Shopify API: use latest version

sterlinghirsh commented:
CR :zap: I left a comment though

sctice commented:
CR :flags:

danielbeardsley commented:
QA :+1: 

Tested again and diffed the order structure before and after this change. Only a new new fields were added to the api result.
